File: Bai_9/main.py
#Cách 1 Dùng vòng lặp For()
def tongqua(n, a):
    sum_val = 0
    for i in range(1, n + 1):
        sum_val += (a * (n - i) + 1)
    return sum_val

n, a = map(int, input().split())
print(tongqua(n, a))

#Cách 2 Dùng tổng của cấp số cộng
print(n*(2+(n-1)*(a))//2)

File: Bai_9/main.cpp
#include <iostream>
using namespace std;
// Cách 1: Sử dụng vòng lặp for
int tongqua(int n, int a) {
    int sum_val = 0;
    for (int i = 1; i <= n; ++i) {
        sum_val += (a * (n - i) + 1);
    }
    return sum_val;
}

int main() {
    int n, a;
    cin >> n >> a;
    cout << tongqua(n, a) << endl;

    // Cách 2: Sử dụng công thức tổng của cấp số cộng
    int sum = n * (2 + (n - 1) * a) / 2;
    cout << sum << endl;

    return 0;
}
